### ISSUE TITLE
Version final variable not resolved

### ISSUE THREAD

**Issue by delanym** [openrewrite/rewrite-maven-plugin#330]
Do this with Maven 3.8.5 if possible

```
mvn archetype:generate -B -DarchetypeGroupId=org.apache.maven.archetypes -DarchetypeArtifactId=maven-archetype-quickstart -DarchetypeVersion=1.1  -DgroupId=com.test -DartifactId=test -Dversion='${revision}${changelist}'

mvn -f test validate -Drevision=1.0.0 -Dchangelist=-SNAPSHOT

mvn -f test org.openrewrite.maven:rewrite-maven-plugin:4.22.2:cyclonedx -Drevision=1.0.0 -Dchangelist=-SNAPSHOT
```

You'll see that the report filename is correct, but the /bom/metadata/component/version hasn't resolved changelist.

Then do it again but add `sha1`.

```
mvn archetype:generate -B -DarchetypeGroupId=org.apache.maven.archetypes -DarchetypeArtifactId=maven-archetype-quickstart -DarchetypeVersion=1.1  -DgroupId=com.test -DartifactId=test -Dversion='${revision}${sha1}${changelist}'

mvn -f test validate -Drevision=1.0.0 -Dsha1=.2f0e86ffe6a1ab55a613a631b18ac42545c5abda -Dchangelist=-SNAPSHOT

mvn -f test org.openrewrite.maven:rewrite-maven-plugin:4.22.2:cyclonedx -Drevision=1.0.0 -Dsha1=.2f0e86ffe6a1ab55a613a631b18ac42545c5abda -Dchangelist=-SNAPSHOT
```

The changelist is now resolved and sha1 is not.

**Comment by tkvangorder** [comment]
This may be because the properties are being defined on the command line. I am looking into this.

**Comment by delanym** [comment]
Thank you. Its not though, if you remember I had the issue with these properties set in ${maven.multiModuleProjectDirectory}/.mvn/maven.config and using the run goal

**Comment by delanym** [comment]
BTW why does this plugin have a goal to generate a BOM? Especially when there's already one available at https://github.com/CycloneDX/cyclonedx-maven-plugin

**Comment by tkvangorder** [comment]
Yeah, I remember this issue.  Properties specified on the command line or via maven.config are added to the system properties by Maven. There are two problems I need to sort out: 

- Asking a resolved pom for its coordinates should resolve any property placeholders (This is the easy part)
- We are using a late-binding approach in getValue(), in that the last thing we do is consult the system properties. This does not impact the maven plugin but does impact execution in the SaaS. Unfortunately, If the sources are serialized and then deserialized into a different VM, the system properties will be different. So I think we need to record the system properties at the time of ingest (when the maven plugin is running)

**Comment by tkvangorder** [comment]
The Cyclonedx was a specific ask from one of our design partners.
